handbook/repository/workspace-settings.md

```diff
@@ -0,0 +1,180 @@
+# Workspace Settings
+
+Various project settings can be controlled from the `workspace.json`
+file, located in the root of the repository.
+
+## E2E Testing configuration for timeouts
+
+We are experiencing timeouts during some of our `Cypress` end-to-end
+tests. This is due to long build and compile time of `NextJS` apps.
+So we have a seperate `e2e-ci` task setup in the `workspace.json`
+configuration of each `e2e` app. With this configuration we can let
+`Cypress` wait until the server is ready before starting it's test
+execution.
+
+### NextJS apps pre-requisites
+
+#### Setting API_MOCKS
+
+We need to make sure that the `API_MOCKS` environment variable is set in
+the `next.config.js`
+
+```javascript
+// ...
+serverRuntimeConfig: {
+  // ...
+},
+publicRuntimeConfig: {
+  // ...
+},
+env: {
+  API_MOCKS: process.env.API_MOCKS,
+},
+```
+
+### React apps pre-requisites
+
+For React apps we first need to add a task `static-serve`
+in the target web app. For example for the `service-portal-e2e`
+it is `service-portal`
+
+```json
+"static-serve": {
+  "builder": "@nrwl/workspace:run-commands",
+  "options": {
+    "commands": [
+      {
+        "command": "yarn serve -l 4200 -s dist/apps/service-portal",
+        "forwardAllArgs": false
+      }
+    ],
+    "readyWhen": "Accepting connections"
```

Comment:
Just curious: Why is `forwardAllArgs` and `readyWhen` needed?

Reply:
When not using `forwardAllArgs` the execution hits this error:
```
ERROR: Unknown or unexpected option: --watch
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Warning: @nrwl/run-commands command "yarn serve -l 4200 -s dist/apps/service-portal --watch=false" exited with non-zero status code
 ```
Nx seems to be adding `--watch=false` to all commands by default, which is failing the `serve` package.

`readyWhen` is needed to continue the process, otherwise `serve` blocks the process.